SkewbSkills/skewbskillsscripts.js

```diff
@@ -1358,13 +1358,39 @@ function ScramblePlusColour() {
         document.getElementById("colourlabel").style.background = 'white';
     }
 }
+function getRandomInt(max) {
+    return Math.floor(Math.random() * max);
+  }
+  
+
 function ScramblePlusColourAlg() {
     if (scramblelistAlg.length == 0) {
         changescrlenAlg();
     }
 
     var scramblezumanzeigenAlg = scramblelistAlg[scramblelistAlg.length - 1];
     scramblelistAlg.pop();
+    // get random int for top site color: 0, 1, 2, 3: /, x, x2, x' | 4, 5: z, z'
+    // get random int for side color: 0, 1, 2, 3: /, y, y2, y'
+    // if x2y2 do z2 instead
+    var firstRotations = ["", "x ", "x2 ", "x' ", "z ", "z' ", "z2 "]
+    var firstRotationsInv = ["", " x'", " x2", " x", " z'", " z", " z2"]
+    var secondRotations = ["", "y ", "y2 ", "y' "]
+    var secondRotationsInv = ["", " y'", " y2", " y"]
+    var firstRotationIndex = getRandomInt(6);
+    var secondRotationIndex = getRandomInt(4);
+    if (firstRotationIndex == 2 && secondRotationIndex == 2) {
+        firstRotationIndex = 6;
+        secondRotationIndex = 0;
+    }
+    var firstRotation = firstRotations[firstRotationIndex];
+    var secondRotation = secondRotations[secondRotationIndex];
+    var firstRotationInv = firstRotationsInv[firstRotationIndex];
+    var secondRotationInv =  secondRotationsInv[secondRotationIndex];
+    var rotations = firstRotation + secondRotation;
+    var backRotations = secondRotationInv + firstRotationInv;
+    scramblezumanzeigenAlg = rotations + scramblezumanzeigenAlg + backRotations;
```

Comment:
I haven't really used or looked at the one-look trainer, but if it too doesn't have randomized layer colors, then yes I think it would be useful there as well.

Reply:
It's very similar to this algorithm trainer, basically only adding few moves on top. If you want, it would be great if this patch could be added there as well to have the different training modes acting in a similar way.